lib/matplotlib/scale.py

```diff
@@ -478,10 +478,111 @@ def get_transform(self):
         return self._transform
 
 
+def _mask_non_logit(a):
+    """
+    Return a Numpy masked array where all values outside ]0, 1[ are
+    masked. If all values are inside ]0, 1[, the original array is
+    returned.
+    """
+    a = a.copy()
+    mask = (a <= 0.0) | (a >= 1.0)
+    a[mask] = np.nan
+    return a
+
+
+def _clip_non_logit(a):
+    a = a.copy()
+    a[a <= 0.0] = 1e-300
+    a[a >= 1.0] = 1 - 1e-300
+    return a
+
+
+class LogitTransform(Transform):
+    input_dims = 1
+    output_dims = 1
+    is_separable = True
+    has_inverse = True
+
+    def __init__(self, nonpos):
+        Transform.__init__(self)
+        if nonpos == 'mask':
+            self._handle_nonpos = _mask_non_logit
+        else:
+            self._handle_nonpos = _clip_non_logit
+        self._nonpos = nonpos
+
+    def transform_non_affine(self, a):
+        """logit transform (base 10), masked or clipped"""
+        a = self._handle_nonpos(a)
+        if isinstance(a, ma.MaskedArray):
```

Comment:
see below: the code for the LogTransforms has the same problem all over the place, I was just copying from there to keep the codebase uniform. But now if mpl is moving away from masked arrays for whatever reason, I'll just correct all the LogTransforms classes too.